### ISSUE TITLE
dialects: merge snitch dialects

### ISSUE THREAD

**Issue by superlopuh** [xdslproject/xdsl#2885]
From experience trying to explain our lowerin in the paper and presentations, it seems that we have too many dialects to explain what's going on. I propose to move everything to two dialects: `snitch` and `riscv_snitch`.

More concretely:

```
snitch_stream.* -> riscv_snitch
snitch.* -> riscv_snitch
memref_stream.* -> snitch
```

**Comment by JosseVanDelm** [comment]
I tend to agree with this sentiment, but I'd check in with @jorendumoulin since he was planning to use a similar lowering approach for the snax streamers which is slightly different.
@jorendumoulin what are your thoughts on this?

**Comment by jorendumoulin** [comment]
hello yes!
For the first two, it definitely makes sense.
For the third, I am finally doing what I said I would be doing a month a ago and using the `memref_stream` for SNAX streamers.

It seems our lowering paths are like this:
```
                    ┌─────────────►  snitch_stream
                    │                             
                    │                             
linalg ────►  memref_stream                       
                                                  
                    │                             
                    │                             
                    └─────────────►  snax_stream  
```

I think the memref_stream abstraction definitely makes sense and is proving to be very useful for me. If merging the dialects means just moving over the operations, then fine (although memref_stream is quite snitch/risc-v-independent, so I'm not sure if this is logical). If it means actually  changing the operations, this would be inconvenient for me.
